blog/content/post/2025-04-14-ram-wipe.md

```diff
@@ -0,0 +1,226 @@
+---
+title: ram-wipe againts RAM attacks
+abstract: 'This post will introduce some of the very popular attacks that target
+electronic devices - the RAM attacks, but the main topic will be the
+verification of ram-wipe software solution protection from the attacks.'
+cover: /covers/ram-wipe.png
+author: daniil.klimuk
+layout: post
+published: true
+date: 2025-04-14
+archives: "2025"
+
+tags:
+  - RAM
+  - wipe
+  - Linux
+  - dracut
+  - cold
+  - boot
+  - attack
+categories:
+  - Security
+
+---
+
+Hello there! This post will introduce some of the very popular attacks that
+target electronic devices - the RAM attacks, but the main topic will be the
+verification of [`ram-wipe` software
+solution](https://github.com/kicksecure/ram-wipe) protection from the attacks.
+
+## Introduction to RAM attacks
+
+Here the focuse is on two types of attack: cold-boot and warm-boot attacks.
+
+Warm-boot attacks focuse on reading data left in RAM by victim's operating
+system or application using attacker's operating system or application without
+disconnecting power from the device. For example, rebooting victim's operating
+system via `systemctl reboot` (without breaking victim's system execution flow)
+and booting some malware to dump or search RAM.
+
+From [`thecyberwire.com`](https://thecyberwire.com/glossary/cold-boot-attack),
+a cold boot attack is:
+
+> A side-channel attack in which an attacker with physical access to a device
+> retrieves encryption keys after a cold system reboot. A computer is "cold
+> booted" when its power is quickly cycled without the machine shutting down
+> cleanly or, in some cases, by hitting a reset on the device. See also
+> data remanence.
```

Comment:
Wiki also has a lot of good materials: https://en.wikipedia.org/wiki/Cold_boot_attack

There is a big difference in the boot process on simple ARM vs complex ARM/x86/POWER. That why we have ACPI power states: https://en.wikipedia.org/wiki/ACPI#POWER-STATES